application-load-balancer/lambda_function/secret_rotator.py

```diff
@@ -0,0 +1,450 @@
+    # Obtain the pending secret value
+        pending = service_client.get_secret_value(
+            SecretId=arn,
+            VersionId=token,
+            VersionStage=AWSPENDING
+        )
+
+        # Obtain metadata and find the current version
+        metadata = service_client.describe_secret(SecretId=arn)
+        current, currenttoken = None, None
+
+        for version in metadata.get("VersionIdsToStages", {}):
+            if AWSCURRENT in metadata["VersionIdsToStages"].get(version, []):
+                currenttoken = version
+                current = service_client.get_secret_value(
+                    SecretId=arn,
+                    VersionId=currenttoken,
+                    VersionStage=AWSCURRENT
+                )
+                logger.info("Getting current version: %s" % (version))
+                break  # Found AWSCURRENT, exit loop
+
+        if not current:
+            raise ValueError("No AWSCURRENT version found")
+
+        # Parse secrets from JSON format
+        pendingsecret = json.loads(pending['SecretString'])
+        currentsecret = json.loads(current['SecretString'])
+
+        return pendingsecret, currentsecret
+
+    def create_secret(self, service_client, arn, token):
+        """Create the secret
+        This method first checks for the existence of a current secret for the passed in token. If one does not exist, it will generate a
+        new secret and put it with the passed in token.
+        Args:
+            service_client (client): The secrets manager service client
+            arn (string): The secret ARN or other identifier
+            token (string): The ClientRequestToken associated with the secret version
+        Raises:
+            ResourceNotFoundException: If the secret with the specified arn and stage does not exist
+        """
+        # Make sure the current secret exists
+        service_client.get_secret_value(
+            SecretId=arn,
+            VersionStage=AWSCURRENT
+        )
+
+        # Now try to get the secret version, if that fails, put a new secret
+        try:
+            service_client.get_secret_value(
+                SecretId=arn,
+                VersionId=token,
+                VersionStage=AWSPENDING
+            )
+            logger.info("createSecret: Successfully retrieved secret")
+
+        except service_client.exceptions.ResourceNotFoundException:
+
+            # Generate a random password
+            passwd = service_client.get_random_password(
+                ExcludePunctuation=True
+            )
+            service_client.put_secret_value(
+                SecretId=arn,
+                ClientRequestToken=token,
+                SecretString='{\"HEADERVALUE\":\"%s\"}' % passwd['RandomPassword'],
+                VersionStages=['AWSPENDING']
+            )
+            logger.info("createSecret: Successfully put secret for version %s" % (token))
```

Comment:
general comment: prefer f-string in python:
f"createSecret: Successfully put secret for version {token}"

Reply:
Changed to f string